OpenRA.Mods.Common/Traits/Upgrades/DisableOnUpgrade.cs

```diff
@@ -18,12 +18,13 @@ public class DisableOnUpgradeInfo : UpgradableTraitInfo
 		public override object Create(ActorInitializer init) { return new DisableOnUpgrade(this); }
 	}
 
-	public class DisableOnUpgrade : UpgradableTrait<DisableOnUpgradeInfo>, IDisable, IDisableMove
+	public class DisableOnUpgrade : UpgradableTrait<DisableOnUpgradeInfo>, IDisable, IDisableMove, IPowerModifier
 	{
 		public DisableOnUpgrade(DisableOnUpgradeInfo info)
 			: base(info) { }
 
 		public bool Disabled { get { return !IsTraitDisabled; } }
 		public bool MoveDisabled(Actor self) { return !IsTraitDisabled; }
+		public int GetPowerModifier() { return IsTraitDisabled ? 100 : 0; }
```

Comment:
As conditions (upgrades) become more widely used, `DisableOnUpgrade` might become redundant and probably will be removed eventually.